testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java

```diff
@@ -53,4 +54,9 @@ public boolean isCassandra4() {
     return !backend.isDse()
         && Version.parse(backend.clusterVersion()).nextStable().compareTo(Version.V4_0_0) >= 0;
   }
+
+  public static Instant now() {
+    // Avoid using Instants with nanosecond precision as nanos may be lost on the server side
```

Comment:
In the past during Jepsen tests, there were problems causing by trimming the nano-seconds.
(https://aphyr.com/posts/294-jepsen-cassandra)
Are you sure that the nanoseconds are trimmed down?

Reply:
This is not about C* internal time tracking. This is purely for user-level data.

Yes, I observed the precision loss when the test is executed by Java 11 but Stargate run in Java 8.

Please note, though, that we currently run tests using Java 8 in CI, in which case test data uses `Instant` objects with millisecond precision, AFAIK.

We can certainly open a follow-up issue for this.

Reply:
Ok. let's create a follow-up ticket.

Reply:
#521